### ISSUE TITLE
I can edit attendance records for the kids in my care

### ISSUE THREAD

**Issue by csprayregen** [pieforproviders/pieforproviders#1170]
## ✨ What new feature do you want to see?
I can make changes to the attendance records I have added for the kids in my care (see #1125 for adding records and #1169 for viewing records).

## 🧶 Acceptance Criteria

- [x] I can edit the check in time
- [x] I can edit the check out time
- [x] I can change a status to "absent"
- [x] I can change the reason for an absence
- [x] If a record came from WS, I cannot edit it. I am directed to edit it in WS --> pending discussion of WS attendance in #1169 
- [x] - [ ] I can add a second check in/out time
- [x] If there are multiple check in/out times, I can edit both 
-- talking with @cjhaddad we decided to add these ACs: --
- [ ] ~If I put a check out time that is earlier than my check in time, I will see a message that stops me from doing so, like this: ~
![image](https://user-images.githubusercontent.com/26929739/152246673-ee205086-5422-43e3-9af1-482c8058f9b2.png) --> CS: per 2/7 team meeting @katelovescode to create separate ticket for this
 - [ ] If I haven't entered a check in & check out time OR if I haven't checked Absent, the "Save" button will be disabled --> CS: per 2/7 team meeting, button will be disabled unless I have a check in time OR checked Absences. I do not need a check out time for this validation

## 🖍️ Implementation Details

[**figma link**](https://www.figma.com/file/uCveVQQnYcw3MsenD1PNcG/P4P-2.0?node-id=5556%3A32631)

> **NOTE TO ENGINEERS**
> 
> Please ensure that you are running Google Lighthouse and/or WebAIM (Wave) on any UI elements in your pull request to check for accessibility issues.
> 
> If it is out of scope to address those accessibility issues in your pull request (because the issue is in a parent component and needs refactoring to work, for example), **you are responsible for noting those issues in your pull request and writing bug tickets to address the issues - make sure you link the new tickets in your pull request**

**Comment by katelovescode** [comment]
Blocked by #1312

**Comment by rebeccakarasiktw** [comment]
Per conversation with @cjhaddad, adding to scope of this card:
* Update to AC so that the buttons say "Save" and "Cancel"
* "Save" button should be disabled until both check in / check out are present OR you choose absent
* Add error message w/in modal if check-out time is earlier than check-in time.

TO THINK ABOUT MORE 
* How do we best communicate to the user once the updates have been made and saved. Some kind of message to confirm your save went thru?

**Comment by csprayregen** [comment]
CS QA:
- [ ] I can edit the check in time
- [ ]  I can edit the check out time
- [ ]  I can change a status to "absent"
- [ ]  I can change the reason for an absence
- [ ]  If a record came from WS, I cannot edit it. I am directed to edit it in WS --> pending discussion of WS attendance in https://github.com/pieforproviders/pieforproviders/issues/1169
- [ ]  - [ ] I can add a second check in/out time
- [ ]  If there are multiple check in/out times, I can edit both
- [ ]  If I haven't entered a check in time OR if I haven't checked Absent, the "Save" button will be disabled 
- [ ] Update to AC so that the buttons say "Save" and "Cancel"

**Comment by katelovescode** [comment]
>  If a record came from WS, I cannot edit it. I am directed to edit it in WS --> @katelovescode how do you suggest I QA this one?

Hmmm.  Looking at staging, this is actually a bug.  All 4 of these kids have Wonderschool IDs, so you shouldn't be able to edit them.  I was going to add a WS ID to show you the disabled edit, but seems like they all already have them.  cc @cjhaddad did we lose this behavior somewhere along the way?

**Comment by katelovescode** [comment]
> > If a record came from WS, I cannot edit it. I am directed to edit it in WS --> @katelovescode how do you suggest I QA this one?
> 
> Hmmm. Looking at staging, this is actually a bug. All 4 of these kids have Wonderschool IDs, so you shouldn't be able to edit them. I was going to add a WS ID to show you the disabled edit, but seems like they all already have them. cc @cjhaddad did we lose this behavior somewhere along the way?

Meanwhile, I took the WS ID off of Manuel so when we get this sorted out, you should be able to edit Manuel's attendances and nobody else's.

**Comment by katelovescode** [comment]
Okay Candice, after submitting my bugfixes, here's what I'm still seeing.

Outstanding issues:
- [ ] If a record came from WS, I cannot edit it. I am directed to edit it in WS --> I am able to edit all kids, despite Kate's note below that I should only be able to edit one
<img width="770" alt="Screen Shot 2022-02-24 at 9 49 07 PM" src="https://user-images.githubusercontent.com/8364647/155650165-cc651be7-08d6-4c98-9d19-c3b96e2b6f14.png">
Manuel should be the only one who can be edited, but on the edit page, I see edit icons on all of the kids:
<img width="1292" alt="Screen Shot 2022-02-24 at 9 50 14 PM" src="https://user-images.githubusercontent.com/8364647/155650231-16a9b33e-1193-4792-ba0f-16c5e33ef6ba.png">

- [ ] I can change a status to "absent" --> doesn't seem to be saving - see screenshots
Absences seem to be saving properly now.

**Comment by csprayregen** [comment]
CS QA: 

- [ ] I can edit the check in time
- [ ]  I can edit the check out time
- [ ]  I can change a status to "absent"
- [ ]  I can change the reason for an absence
- [ ]  If a record came from WS, I cannot edit it. I am directed to edit it in WS
- [ ]  I can add a second check in/out time
- [ ]  If there are multiple check in/out times, I can edit both
- [ ]  If I haven't entered a check in time OR if I haven't checked Absent, the "Save" button will be disabled

**Comment by csprayregen** [comment]
CS QA
- [ ]  I can edit the check in time
- [ ]  I can edit the check out time
- [ ]  I can change a status to "absent"
- [ ]  If a record came from WS, I cannot edit it. I am directed to edit it in WS
- [ ]  I can add a second check in/out time
- [ ]  If there are multiple check in/out times, I can edit both
- [ ]  If I haven't entered a check in time OR if I haven't checked Absent, the "Save" button will be disabled

**Comment by csprayregen** [comment]
CS QA
- [ ] I can edit the check in time
- [ ]  I can edit the check out time
- [ ]  I can change a status to "absent"
- [ ]  If a record came from WS, I cannot edit it. I am directed to edit it in WS
- [ ]  I can add a second check in/out time
- [ ]  If there are multiple check in/out times, I can edit both
- [ ]  If I haven't entered a check in time OR if I haven't checked Absent, the "Save" button will be disabled

**Comment by csprayregen** [comment]
@cjhaddad same QA is still not passing. I was able to successfully change a record to Absent for March 7, but for March 6 it won't change when I check the box. Unlike before, it's not updated in the dialogue box or in the View Attendance page. The "edit" doesn't seem to be sticking. See screenshots. 
<img width="314" alt="Screen Shot 2022-03-07 at 1 13 45 PM" src="https://user-images.githubusercontent.com/26717304/157102209-f7a1e865-8ada-4b67-b26c-5a7ffa7d91ed.png">
<img width="393" alt="Screen Shot 2022-03-07 at 1 13 55 PM" src="https://user-images.githubusercontent.com/26717304/157102213-66b7fb6d-259b-48ac-a38d-613db20c8fc0.png">
